[Xrd] Wait for the current PollE poll to finish after removing a Link

smithdh opened the pull request:
This is a change to avoid issue #1928.

smithdh committed:
[Xrd] Wait for the current PollE poll to finish after removing a Link

abh3 commented on src/Xrd/XrdPollE.icc:
```diff
@@ -79,9 +88,30 @@ XrdPoll *XrdPoll::newPoller(int pollid, int maxfd)
-  
+
+/******************************************************************************/
+/*                              A d d E v e n t F d                           */
```
This should be AddWaitFD (looks like you must have changed your mind at some point).

abh3 commented on src/Xrd/XrdPollE.icc:
```diff
@@ -164,6 +194,44 @@ void XrdPollE::Exclude(XrdPollInfo &pInfo)
+}
+
+/******************************************************************************/
+/*                          H a n d l e E v e n t F d                         */
```
This should be HandleWaitFD

abh3 commented on src/Xrd/XrdPollE.icc:
```diff
@@ -272,9 +351,28 @@ void XrdPollE::Start(XrdSysSemaphore *syncsem, int &retcode)
 }
 
+/******************************************************************************/
+/*                             W a i t E v e n t F d                          */
```
This should ne WaitPoller (perhaps a better name would be Wait4Poller)

abh3 reviewed (changes requested):
Just comment changes are needed. Presumably the code was tested with the "sleeps" that caused it to fail and now it doesn't fail, right? The only other suggestion is consider repackaging the "eventfd' processing into a separate class so that it can be used elsewhere should the need arise.

smithdh commented:
Thanks very much for the comments. I've changed those; the banners and renamed Wait4Poller. I've tried it with my "sleeps" it solves what I could provoke. (I don't know how to reproduce the original production-system problem, as that seems to be quite rare).

Repackaging as a separate class is likely a better way to make it. However I'm not sure I can find a nice general interface for it. (It would probably be used with an epoll-poller, as eventfd is linux specific like epoll). I don't think we have the same problem in XrdSysIOEventsPollE because of how that poller works. So I propose to leave it here (not separated into another class) now.
